### PR TITLE
[stable/mysql] Adding BYO SSL certificates option

### PR DESCRIPTION
**Pull request by alika**
**What this PR does / why we need it**: This PR adds the option to setup SSL for MySQL connections by allowing the chart user to specify the CA certificate, server certificate, and server key in values file. 

**Special notes for your reviewer**: 

- I've also added a helm test to verify the SSL connection is setup correctly only when SSL has been enabled.
- To enable SSL you also need to enable a custom configuration in the `configurationFiles` section

### CODE REVIEW

File: stable/mysql/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: mysql
-version: 0.3.6
+version: 0.3.7
```
Comment (timstoop) [inline stable/mysql/Chart.yaml]: You're adding new features, please bump the minor version of the Chart.

**Review by timstoop — CHANGES_REQUESTED** [review]

**Comment by timstoop** [comment]
Please rebase and bump the version again, another commit got merged.

**Comment by alika** [comment]
Apologies @timstoop. I seem to have really messed up this rebase. Is there an easy way to undo this or will this be easier to create a new PR?

**Comment by mattfarina** [comment]
/cc @viglesiasce

because you're the maintainer

File: stable/mysql/README.md [hunk 2]
```diff
@@ -71,6 +71,10 @@ The following table lists the configurable parameters of the MySQL chart and the
 | `persistence.subPath`                | Subdirectory of the volume to mount       | `nil`                                                |
 | `resources`                          | CPU/Memory resource requests/limits       | Memory: `256Mi`, CPU: `100m`                         |
 | `configurationFiles`                 | List of mysql configuration files         | `nil`                                                |
+| `ssl.enabled`                        | Setup and use SSL for MySQL connections   | `false`                                              |
+| `ssl.ca`                             | CA certificate                            | `nil`                                                |
+| `ssl.cert`                           | Server certificate (public key)           | `nil`                                                |
+| `ssl.key`                            | Server key (private key)                  | `nil`                                                |
```
Comment (mattfarina) [inline stable/mysql/README.md]: The wordpress chart is an example of one that does ssl and secrets with a slightly different pattern ([you can see it here](https://github.com/kubernetes/charts/blob/master/stable/wordpress/values.yaml#L170)). The idea is that setting the secret to use is different from passing in the values and creating a secret. This way you can use an outside secret, not managed, by the chart if you want.

**Comment by alika** [comment]
@mattfarina How does this look now?

**Comment by whereisaaron** [comment]
Hi @alika, thank you very much for add the SSL support to this chart!

This PR also added a helm hook heath test. I personally don't want to also run that hook, as I prefer to leave that health check to the k8s health probes. Also, the current test embeds the MySQL root password in a ConfigMap, which kind of defeats the purpose of keeping it in a Secret 😄 

Could I request you add an `enabled` option to the `values.yaml` to give user the option to disable the test if users don't happen to want it?

**Pull request merged**